src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -0,0 +1,450 @@
+            logger.info(msg)
+            for rse in rses:
+                if rse in doc['currentRSEs']:
+                    doc['currentRSEs'].remove(rse)
+                    modify = True
+                    msg = f"delete rse {rse} from currentRSEs"
+                    report.addEntry('monitoring', uuid, msg)
+
+    # persist an up-to-date version of the pileup data structure in MongoDB
+    if modify:
+        logger.info(f"monitoring task {uuid}, update {pname}")
+        mgr.updatePileup(doc)
+        msg = f"update pileup {pname}"
+        report.addEntry('monitoring', uuid, msg)
+    else:
+        logger.info("monitoring task {uuid}, processed without update")
+
+
+def inactiveTask(doc, spec):
+    """
+    Perform single inactive task over provided MSPileup document
+
+    :param mgr: MSPileupData manager
+    :param doc: MSPileup document
+    :param spec: task spec dict
+    """
+    mgr = spec['manager']
+    uuid = spec['uuid']
+    rucioClient = spec['rucioClient']
+    rucioAccount = spec['rucioAccount']
+    report = spec['report']
+    logger = spec.get('logger', getMSLogger(False))
+    pname = doc['pileupName']
+    if isinstance(rucioClient, Client):
+        inputArgs = {'scope': 'cms', 'name': pname, 'account': rucioAccount}
+        rules = [r for r in rucioClient.list_replication_rules(inputArgs)]
+    else:
+        rules = [r for r in rucioClient.listDataRules(pname)]
+    modify = False
+
+    for rdoc in rules:
+        # make a Rucio call to delete that rule id
+        rid = rdoc['id']
+        msg = f"inactive task {uuid}, container: {pname} for Rucio account {rucioAccount}, delete replication rule {rid}"
+        logger.info(msg)
+        if isinstance(rucioClient, Client):
+            rucioClient.delete_replication_rule(rid)
+            rses = [r['rse'] for r in rucioClient.list_rses(rdoc['rse_expression'])]
+        else:
+            rucioClient.deleteRule(rid)
+            rses = [rse for rse in rucioClient.evaluateRSEExpression(rdoc['rse_expression'])]
+
+        # remove the rule id from ruleIds (if any) and remove the RSE name from currentRSEs (if any)
+        if rid in doc['ruleIds']:
+            doc['ruleIds'].remove(rid)
+            modify = True
+            msg = f"remove rid {rid}"
+            report.addEntry('inactive', uuid, msg)
+        for rse in rses:
+            if rse in doc['currentRSEs']:
+                doc['currentRSEs'].remove(rse)
+                modify = True
+                msg = f"remove rse {rse}"
+                report.addEntry('inactive', uuid, msg)
+
+    # make a log record if the DID + Rucio account tuple does not have any existent rules
+    if not rules:
+        msg = f"inactive task {uuid}, container: {pname} for Rucio account {rucioAccount} does not have any existing rules, proceed without update"
```

Comment:
line too long, please break it in two lines.

Reply:
done